### DISCUSSION TITLE
when the position change of object  from horizontal plane to vertical plane then object rotate, i don't want that rotation

### DISCUSSION THREAD

**Discussion by UmerHamza** [sceneview/sceneview#509] [Q&A]
When I position the object horizontally or vertically, if I then move it and encounter a chair during horizontal movement or a window during vertical movement, the object begins to tilt and alters its rotation.

**Comment by ThomasGorisse** [comment]
When an anchor transitions between horizontal and vertical planes, the model rotates because the anchor's pose includes the plane's orientation. In v4.0.0-rc.1, you can fix the rotation by overriding the node's rotation:

```kotlin
AnchorNode(anchor = anchor) {
    ModelNode(
        modelInstance = model,
        rotation = Rotation(0f, 0f, 0f) // force upright orientation
    )
}
```

Alternatively, lock plane detection to one type:
```kotlin
ARSceneView(
    sessionConfiguration = { _, config ->
        config.planeFindingMode = Config.PlaneFindingMode.HORIZONTAL
    }
)
```
